packages/markdown/processor/parse/magic-block-parser.js

```diff
@@ -0,0 +1,219 @@
+/* eslint-disable consistent-return */
+const RGXP = /^\[block:(.*)\]([^]+?)\[\/block\]/;
+
+const WrapPinnedBlocks = (node, json) => {
+  if (!json.sidebar) return node;
+  return {
+    children: [node],
+    type: 'rdme-pin',
+    data: {
+      hName: 'rdme-pin',
+      className: 'pin',
+    },
+  };
+};
+
+function tokenize(eat, value) {
+  let [match, type, json] = RGXP.exec(value) || [];
+
+  if (!type) return;
+
+  match = match.trim();
+  type = type.trim();
+  json = (json && JSON.parse(json)) || {};
+
+  switch (type) {
+    case 'code': {
+      const children = json.codes.map(obj => ({
+        type: 'code',
+        value: obj.code,
+        meta: obj.name || null,
+        lang: obj.language,
+        className: 'tab-panel',
+        data: {
+          hName: 'code',
+          hProperties: {
+            meta: obj.name || null,
+            lang: obj.language,
+          },
+        },
+      }));
+      if (children.length === 1) return eat(match)(WrapPinnedBlocks(children[0], json));
+      return eat(match)(
+        WrapPinnedBlocks(
+          {
+            children,
+            className: 'tabs',
+            data: { hName: 'code-tabs' },
+            type: 'code-tabs',
+          },
+          json,
+        ),
+      );
+    }
+    case 'api-header': {
+      return eat(match)(
+        WrapPinnedBlocks(
+          {
+            type: 'heading',
+            depth: json.level || 2,
+            children: this.tokenizeInline(json.title, eat.now()),
+          },
+          json,
+        ),
+      );
+    }
+    case 'image': {
+      return eat(match)(
+        WrapPinnedBlocks(
+          json.images.map(img => {
+            const [url, title] = img.image;
+            return {
+              type: 'image',
+              url,
+              title,
+              alt: img.caption,
+              data: {
+                hProperties: {
+                  caption: img.caption,
+                },
+              },
+            };
+          })[0],
+          json,
+        ),
+      );
+    }
+    case 'callout': {
+      json.type = {
+        // @todo: I should probably just be using the original class names here...
```

Comment:
Which original class names?

Reply:
This comment should've been removed a while back. It was in reference to the old magic block callout classes. (At the end of the day it made more sense to update these classes anyways, so as to avoid our old CSS bleeding in to the new markdown display styles.)